pandas/core/indexes/base.py

```diff
@@ -1395,6 +1395,8 @@ def _format_with_header(self, header: list[str_t], na_rep: str_t) -> list[str_t]
         from pandas.io.formats.format import format_array
 
         values = self._values
+        if is_string_dtype(values.dtype):
+            values = np.asarray(values)
 
         if is_object_dtype(values.dtype):
```

Comment:
Or

```suggestion

        if is_object_dtype(values.dtype) or is_string_dtype(values.dtype):
```

could also be a way to be explicit that string dtypes take this route